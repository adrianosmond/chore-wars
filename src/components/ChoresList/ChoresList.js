import React from 'react';
import FlipMove from 'react-flip-move';

import Chore from 'components/Chore';

import './ChoresList.css';

const ChoresList = ({
  chores, user, game, allChores,
}) => {
  const now = new Date().getTime();
  return (
    <ul className="chores-list">
      { chores.length === 0 ? (
        <div>
          <p className="hide--large">
            You can add chores or customise your character from the menu below
            - tap the arrow to open it.
          </p>
        </div>
      ) : null }
      <FlipMove>
        {chores.filter(chore => now - chore.lastDone >= 60000).map(chore => (
          <li className="chores-list__item" key={chore.title}>
            <Chore chore={chore} allChores={allChores} user={user} game={game} />
          </li>
        ))}
      </FlipMove>
    </ul>
  );
};

export default ChoresList;
