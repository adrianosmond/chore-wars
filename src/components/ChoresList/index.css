.chores-list {
  margin-top: 1rem;
  margin-bottom: 2.5rem;
  padding: 0;
  list-style: none;
}

.chores-list__item {
  margin-top: 1rem;
  max-width: 500px;
  margin-left: auto;
  margin-right: auto;
}

.chores-list__item:first-child {
  margin-top: 0;
}

@media screen and (min-width: 37.5em) {
  .chores-list {
    margin-bottom: 0;
  }
}