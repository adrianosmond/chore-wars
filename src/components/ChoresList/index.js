import React from 'react';
import FlipMove from 'react-flip-move';

import Chore from '../Chore';

import './index.css';

const ChoresList = ({ chores }) => (
  <ul className="chores-list">
    <FlipMove>
    {chores.map(chore =>
      <li className="chores-list__item" key={chore.title}>
          <Chore chore={chore} />
      </li>)}
    </FlipMove>
  </ul>
);

export default ChoresList;
