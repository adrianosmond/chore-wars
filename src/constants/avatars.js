export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
