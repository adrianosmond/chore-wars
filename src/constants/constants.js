const TIME_UNIT = 1000 * 60 * 60 * 24; // 1 day
const MIN_CHORE_FREQUENCY = 1;
const MAX_CHORE_FREQUENCY = 365;
const MIN_CHORE_POINTS = 1;
const MAX_CHORE_POINTS = 200;

const DATE_FORMAT = 'DD/MM/YYYY';

const MAX_POINT_DIFFERENCE = 500;

const MAX_NAME_LENGTH = 10;
const JOIN_CODE_LENGTH = 8;

const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};

const ActionTypes = {
  addChore: 'ADD_CHORE',
  resetChoreDoneDate: 'RESET_CHORE_DONE_DATE',
  blockChore: 'BLOCK_CHORE',
  unblockChore: 'UNBLOCK_CHORE',
  removeChore: 'REMOVE_CHORE',
  updateChore: 'UPDATE_CHORE',
  setChores: 'SET_CHORES',
  makeChain: 'MAKE_CHAIN',
  breakChain: 'BREAK_CHAIN',
  addPoints: 'ADD_POINTS',
  claimPrize: 'CLAIM_PRIZE',
  paidDebt: 'PAID_DEBT',
  setPoints: 'SET_POINTS',
  updateUser: 'UPDATE_USER',
  setAuthUser: 'SET_AUTH_USER',
  setGame: 'SET_GAME',
  setPointsLoaded: 'SET_POINTS_LOADED',
  setChoresLoaded: 'SET_CHORES_LOADED',
  saveStatePostUndo: 'SAVE_STATE_POST_UNDO',
};

export {
  TIME_UNIT,
  MIN_CHORE_FREQUENCY,
  MAX_CHORE_FREQUENCY,
  MIN_CHORE_POINTS,
  MAX_CHORE_POINTS,
  MAX_NAME_LENGTH,
  JOIN_CODE_LENGTH,
  DATE_FORMAT,
  MAX_POINT_DIFFERENCE,
  ActionTypes,
  DefaultAvatar,
};
