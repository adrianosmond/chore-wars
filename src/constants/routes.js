export const LOGIN = '/'
export const CHORES = '/chores'
export const NEW_CHORE = '/chores/new'
