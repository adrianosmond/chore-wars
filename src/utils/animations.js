export const onAppear = (el) => {
  setTimeout(() => {
    el.classList.add('fadeIn');
    setTimeout(() => {
      // eslint-disable-next-line no-param-reassign
      el.style.opacity = '';
      el.classList.remove('fadeIn');
    }, 500);
  }, 0);
};

export const onExit = (el, _, removeElement) => {
  setTimeout(() => {
    el.classList.add('fadeOut');
    setTimeout(removeElement, 500);
  }, 0);
};
