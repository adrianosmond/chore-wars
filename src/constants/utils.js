import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
