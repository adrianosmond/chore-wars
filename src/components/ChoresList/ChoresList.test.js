import React from 'react';
import { shallow } from 'enzyme';
import { processChore } from 'constants/utils';

import ChoresList from './ChoresList';

describe('ChoresList', () => {
  it('Renders an empty list', () => {
    const component = shallow(<ChoresList chores={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('Renders a list', () => {
    const now = new Date('2018-02-20').getTime();
    const list = [{
      ...processChore({
        frequency: 1,
        lastDone: new Date('2018-02-18').getTime(),
        pointsPerTime: 100,
        title: 'Chore 1',
      }, 'chore-1', now),
    }, {
      ...processChore({
        frequency: 0,
        lastDone: new Date('2018-02-19').getTime(),
        pointsPerTime: 10,
        title: 'Chore 2',
      }, 'chore-2', now),
    }];

    const component = shallow(<ChoresList chores={list} />);
    expect(component).toMatchSnapshot();
  });
});
