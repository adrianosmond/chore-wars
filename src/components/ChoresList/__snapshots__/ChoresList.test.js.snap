// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoresList Renders a list 1`] = `
<ul
  className="chores-list"
>
  <FlipMovePropConverter
    delay={0}
    disableAllAnimations={false}
    duration={350}
    easing="ease-in-out"
    enterAnimation="elevator"
    getPosition={[Function]}
    leaveAnimation="elevator"
    maintainContainerHeight={false}
    staggerDelayBy={0}
    staggerDurationBy={0}
    typeName="div"
    verticalAlignment="top"
  >
    <li
      className="chores-list__item"
      key="Chore 1"
    >
      <Chore
        allChores={Object {}}
        chore={
          Object {
            "currentPoints": 200,
            "due": 1518998400000,
            "frequency": 1,
            "lastDone": 1518912000000,
            "percentage": 100,
            "pointsPerTime": 100,
            "slug": "chore-1",
            "title": "Chore 1",
          }
        }
        game="game"
        user="test-user"
      />
    </li>
    <li
      className="chores-list__item"
      key="Chore 2"
    >
      <Chore
        allChores={Object {}}
        chore={
          Object {
            "currentPoints": 10,
            "due": 1518998400000,
            "frequency": 0,
            "lastDone": 1518998400000,
            "percentage": 100,
            "pointsPerTime": 10,
            "slug": "chore-2",
            "title": "Chore 2",
          }
        }
        game="game"
        user="test-user"
      />
    </li>
  </FlipMovePropConverter>
</ul>
`;

exports[`ChoresList Renders an empty list 1`] = `
<ul
  className="chores-list"
>
  <div>
    <p
      className="hide--large"
    >
      You can add chores or customise your character from the menu below - tap the arrow to open it.
    </p>
  </div>
  <FlipMovePropConverter
    delay={0}
    disableAllAnimations={false}
    duration={350}
    easing="ease-in-out"
    enterAnimation="elevator"
    getPosition={[Function]}
    leaveAnimation="elevator"
    maintainContainerHeight={false}
    staggerDelayBy={0}
    staggerDurationBy={0}
    typeName="div"
    verticalAlignment="top"
  />
</ul>
`;
